odoo/fields.py

```diff
@@ -4584,6 +4584,11 @@ def browse(ids):
 
         if self.store:
             inverse = self.inverse_name
+            to_inverse = defaultdict(OrderedSet)
+            def flush_to_inverse():
+                for record, line_ids in to_inverse.items():
+                    browse(line_ids)[inverse] = record
+                to_inverse.clear()
```

Comment:
This data structure is not appropriate, because the same line id may be present several times in the dict.
Consider the list of commands `[Command.link(42), Command.unlink(42), Command.link(42)]`.
The dict `to_inverse` will eventually contain, in order: `{record: {42}, False: {42}]`.
Once you flush it, the line with id 42 will not be linked to `record`!

IMHO you should use a direct mapping, and inverse it when flushing:
```suggestion
            to_assign = {}  # {real_line_id: inverse_field_value}
            def flush():
                for value, line_ids in groupby(to_assign, to_assign.__getitem__):
                    browse(line_ids)[inverse] = record
                to_assign.clear()
```